cpp/include/raft/neighbors/detail/knn_brute_force.cuh

```diff
@@ -237,6 +247,27 @@ void tiled_brute_force_knn(const raft::resources& handle,
         }
       }
 
+      if (filter_bitmap != nullptr) {
+        auto distances_ptr          = temp_distances.data();
+        auto count                  = thrust::make_counting_iterator<IndexType>(0);
+        ElementType masked_distance = select_min ? std::numeric_limits<ElementType>::infinity()
+                                                 : std::numeric_limits<ElementType>::lowest();
+        thrust::for_each(resource::get_thrust_policy(handle),
```

Comment:
can we do `raft::linalg::map` here or `map_offset` instead of a `for_each`? Using raft primitives wherever possible allows us to manage their upkeep without regards for the underlying dependencies.

Reply:
The 'map_offset' logic always writes back to HBM even if not prefiltered out, and I worried the performance would get worse..